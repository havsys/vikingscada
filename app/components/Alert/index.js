import React from 'react';
import PropTypes from 'prop-types';

/**
 * Alert
 */
export class Alert extends React.Component {
  render() {
    const {
      iconClass, text, type
    } = this.props;
    return (
      <div className={`alert alert-${type} m-alert m-alert--icon m-alert--air m-alert--square m--margin-bottom-30`} role="alert">
        <div className="m-alert__icon">
          <i className={iconClass} />
        </div>
        <div className="m-alert__text">
          {text}
        </div>
      </div>
    );
  }
}

Alert.propTypes = {
  /**
   * Message
   */
  text: PropTypes.string,

  /**
   * Icon
   */
  iconClass: PropTypes.string,

  /**
   * Alert's type
   */
  type: PropTypes.string
};

Alert.defaultProps = {
  text: '',
  iconClass: '',
  type: 'danger'
};

export default Alert;
