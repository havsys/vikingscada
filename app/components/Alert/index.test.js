import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Alert from './index';

import Adapter from 'enzyme-adapter-react-16';

import renderer from 'react-test-renderer';


describe('>>>ALERT --- Snapshot',()=>{
  it('+++capturing Snapshot of Alert', () => {
    // Render a checkbox with label in the document

    const type = 'danger';
    const iconClass = 'flaticon-exclamation-1';
    const text = 'You are rejected. Please, contact support';

    const alert = renderer.create(
      <Alert type={type} iconClass={iconClass} text={text} />
    );

    let tree = alert.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('>>>ALERT --- Renders Alert message to user',()=>{
  let wrapper = [];
  const type = 'danger';
  const iconClass = 'flaticon-exclamation-1';
  const text = 'You are rejected. Please, contact support';

  Enzyme.configure({ adapter: new Adapter() })

  beforeEach(()=>{
      wrapper = shallow(<Alert type={type} iconClass={iconClass} text={text} />) 
  })

  it('+++ render the DUMB component', () => {
     expect(wrapper.length).toEqual(1)
  });
    
  it('+++ contains text', () => {
      expect(wrapper.find('.m-alert__text').text()).toEqual(text)
  });

  it('+++ has class type', () => {
    expect(wrapper.find('.alert').hasClass(type))
  });

  it('+++ has class iconClass', () => {
    expect(wrapper.find('.m-alert__icon > i').hasClass(iconClass))
  });
  
});