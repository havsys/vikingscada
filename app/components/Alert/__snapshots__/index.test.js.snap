// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ALERT --- Snapshot +++capturing Snapshot of Alert 1`] = `
<div
  className="alert alert-danger m-alert m-alert--icon m-alert--air m-alert--square m--margin-bottom-30"
  role="alert"
>
  <div
    className="m-alert__icon"
  >
    <i
      className="flaticon-exclamation-1"
    />
  </div>
  <div
    className="m-alert__text"
  >
    You are rejected. Please, contact support
  </div>
</div>
`;
